casa/Logging/LogOrigin.cc

```diff
@@ -224,14 +212,12 @@ ostream &operator<<(ostream &os, const LogOrigin &origin)
     return os;
 }
 
-// Goes with the following function. Not static function level because
-// causes problems with our exception emulation.
-static SourceLocation permanent;
-const SourceLocation *SourceLocation::canonicalize(const char *file, Int line)
+SourceLocation SourceLocation::canonicalize(const char *file, Int line)
 {
-    permanent.fileName = file;
-    permanent.lineNumber = line;
-    return &permanent;
+  SourceLocation permanent;
+  permanent.fileName = file;
+  permanent.lineNumber = line;
+  return permanent;
```

Comment:
Since `permanent` is no longer `static`, it's no longer permanent. Please use a different name.
```suggestion
  SourceLocation location;
  location.fileName = file;
  location.lineNumber = line;
  return location;
```